src/fragments/lib/auth/android/signin_next_steps/30_confirm_custom_challenge.mdx

```diff
@@ -94,6 +94,14 @@ RxAmplify.Auth.confirmSignIn(
         );
 ```
 </Block>
+</BlockSwitcher>
 
+<Callout warning>
 
-</BlockSwitcher>
+<b>Special Handling on ConfirmSignIn</b>
+
+During a confirmSignIn call if `failAuthentication=true` is returned by the lambda the session of the request gets invalidated by cognito, a NotAuthorizedException is returned and a new signIn call is expected via Amplify.Auth.signIn
+```java
```

Comment:
do you need a Kotlin variant here for this code snippet?

Reply:
No this is just the exception and is kotlin/java agnostic.

Reply:
Got it.